### ISSUE TITLE
Subgraph endpoint update required

### ISSUE THREAD

**Issue by alex-pakalniskis** [graphprotocol/allocation-optimizer#52]
Hey team,

We've noticed that you're referencing a subgraph deployed on The Graph's hosted service in this repository. **Just a heads-up: after June 12th, hosted service subgraph endpoints will no longer be available as the offering is being deprecated.**

If you are the subgraph owner or maintainer, it's time to upgrade your subgraph to the network. This ensures everything keeps running smoothly and you get access to all the latest features and improvements. Here is a [quick guide](https://thegraph.com/docs/en/sunrise/#how-do-i-upgrade-my-hosted-service-subgraph) with all the upgrade steps.

If you're not the subgraph owner or maintainer, check [Graph Explorer](https://thegraph.com/explorer) to see if the subgraph development team has already upgraded to the network. If you don’t find an upgraded subgraph, a friendly nudge to the subgraph development team would be greatly appreciated—it's a quick process that benefits everyone. Here's the [upgrade guide](https://thegraph.com/docs/en/sunrise/#how-do-i-upgrade-my-hosted-service-subgraph) with all the necessary steps for the subgraph development team.

Once the subgraph is upgraded to the network, you can [create an API key and updated query URL](https://thegraph.com/docs/en/querying/querying-the-graph/) in Subgraph Studio then update this repository to query from the new endpoint, `https://api.studio.thegraph.com/query/<ID>/<SUBGRAPH_NAME>/<VERSION>`

Need more support or have more questions? Feel free to reach out to felix@edgeandnode.com. We're here to help!

Cheers,

Paka

**Comment by alex-pakalniskis** [comment]
Files requiring an update: https://github.com/search?q=repo%3Agraphprotocol%2Fallocation-optimizer%20api.thegraph.com%2Fsubgraphs%2Fname&type=code

**Comment by dyasny** [comment]
@alex-pakalniskis does this mean I have to create my own endpoint to query with allocation optimizer, or will a new endpoint be provided eventually? If I need to create my own endpoint I would appreciate a guide specific for allocation-optimizer, so I can continue reallocating.

**Comment by dyasny** [comment]
I think I have it figured out, I set 
`network_subgraph_endpoint = "https://gateway-arbitrum.network.thegraph.com/api/{API_KEY}/subgraphs/id/{GRAPH_NETWORK_ARBITRUM_SUBGRAPH_ID}"` and the `AllocateOpt` at least didn't crash :) 

The subgraph I used is `DZz4kDTdmzWLWsV373w2bSmoar3umKKH9y82SUKr5qmp` I hope it's correct

**Comment by alex-pakalniskis** [comment]
Yes that looks good to me. Glad it worked (I expected it to be plug-and-play replace old url with new one including API key). Let us know if you run into any issues (we'll update the docs for this repo).

**Comment by dyasny** [comment]
@alex-pakalniskis actually I ran into additional issues trying to reallocate. I have the following settings for the indexer-agent:

```yaml
arbitrumNetwork: arbitrum-one
arbitrumGateway: "https://gateway-arbitrum.network.thegraph.com"
arbitrumSubgraphEndpoint: "https://api.thegraph.com/subgraphs/name/graphprotocol/graph-network-arbitrum"
arbitrumEpochEndpoint: "https://api.thegraph.com/subgraphs/name/graphprotocol/arbitrum-epoch-block-oracle"
```

Do any of these need to be also updated? When trying to close allocations today I ran into https://github.com/graphprotocol/indexer/blob/main/docs/errors.md#ie069 for all of them
